aligner/tests/test_cli.py

```diff
@@ -0,0 +1,112 @@
+
+    def test_align_empty_file(self):
+        with self.subTest("empty file"):
+            result = self.runner.invoke(app, ["align", os.devnull, os.devnull])
+            self.assertNotEqual(result.exit_code, 0)
+            self.assertIn("is empty", result.stdout)
+
+        with self.subTest("file with only empty lines"):
+            with tempfile.TemporaryDirectory() as tmpdir:
+                textfile = os.path.join(tmpdir, "emptylines.txt")
+                with open(textfile, "w", encoding="utf8") as f:
+                    f.write("\n \n   \n")
+                result = self.runner.invoke(app, ["align", textfile, os.devnull])
+                self.assertNotEqual(result.exit_code, 0)
+                self.assertIn("is empty", result.stdout)
+
+    def fetch_ras_test_file(self, filename, outputdir):
+        from urllib.request import Request, urlopen
+
+        repo, path = "https://github.com/ReadAlongs/Studio/", "/test/data/"
+        request = Request(repo + "raw/refs/heads/main" + path + filename)
+        request.add_header("Referer", repo + "blob/main" + path + filename)
+        response = urlopen(request)
+        with open(os.path.join(outputdir, filename), "wb") as f:
+            f.write(response.read())
+
+    def test_align_something(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            tmppath = Path(tmpdir)
+            self.fetch_ras_test_file("ej-fra.txt", tmpdir)
+            txt = tmppath / "ej-fra.txt"
+            self.fetch_ras_test_file("ej-fra.m4a", tmpdir)
+            m4a = tmppath / "ej-fra.m4a"
+            wav = tmppath / "ej-fra.wav"
+            # Under most circumstances, align can take a .m4a input file, but not
+            # in CI. Since it's not a hard requirement, so just convert to .wav.
+            subprocess.run(["ffmpeg", "-i", m4a, wav], capture_output=True)
+            # os.system("ls -la " + tmpdir)
+            textgrid = tmppath / "ej-fra-16000.TextGrid"
+            wav_out = tmppath / "ej-fra-16000.wav"
+
+            with self.subTest("ctc-segmenter align"):
+                result = self.runner.invoke(app, ["align", str(txt), str(wav)])
+                if result.exit_code != 0:
+                    os.system("ls -la " + tmpdir)
+                    print(result.stdout)
+                self.assertEqual(result.exit_code, 0)
+                self.assertTrue(textgrid.exists())
+                self.assertTrue(wav_out.exists())
+
+            with self.subTest("ctc-segmenter extract"):
+                result = self.runner.invoke(
+                    app, ["extract", str(textgrid), str(wav_out), str(tmppath / "out")]
+                )
+                if result.exit_code != 0:
+                    print(result.stdout)
+                self.assertEqual(result.exit_code, 0)
+                self.assertTrue((tmppath / "out/metadata.psv").exists())
+                for i in range(7):
+                    self.assertTrue((tmppath / f"out/wavs/segment{i}.wav"))
```

Comment:
Why `7`?  Is this specific to that input file/output file or is `7` the expected value for all inputs?

Reply:
The file has seven lines.
But good question: it'll be more obvious if I count the lines in the file instead of hard-coding the value.

Reply:
7 non-empty lines, that is...